### PR TITLE
Setup for addition of dashboard self-hosted command

### PR DESCRIPTION
**Pull request by willdavsmith**
# Description

Making necessary changes to current CLI to allow for later pull request adding `dapr dashboard` command in self-hosted mode.

* Updated pkg/standalone.go/untar function to untar all files in tarball rather than just binary
* Added check for kubernetes mode in pkg/dashboard.go
* Added 'githubRepo' as a parameter to installBinary function
* Removed extra error message

## Issue reference

References #407 
References #238 

Will close in next PR

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: pkg/standalone/standalone.go [hunk 1]
```diff
@@ -612,41 +612,45 @@ func untar(filepath, targetDir, binaryFilePrefix string) (string, error) {
...
 			}
+			continue
+		}
 
-			// #nosec G110
-			if _, err := io.Copy(f, tr); err != nil {
-				return "", err
-			}
-			f.Close()
+		f, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
```
Comment (shalabhms) [inline pkg/standalone/standalone.go:637]: these permissions are different than earlier change,,os.O_RDWR missing..? Do we need this additional permissions ?

File: pkg/standalone/standalone.go [hunk 2]
```diff
@@ -612,41 +612,45 @@ func untar(filepath, targetDir, binaryFilePrefix string) (string, error) {
...
 
+	foundBinary := ""
 	for {
 		header, err := tr.Next()
 
-		switch {
-		case err == io.EOF:
-			return "", fmt.Errorf("file is empty")
-		case err != nil:
+		if err == io.EOF {
```
Comment (shalabhms) [inline pkg/standalone/standalone.go:619]: We can still have switch block...no need to change to if/else blocks...

Comment (willdavsmith) [reply pkg/standalone/standalone.go:619]: Actually, it's kind of surprising but this does not work with the switch statement. the 'break' statement will break out of the switch statement rather than the for. My options were to either do this or create a label for the outer for loop and break out of that, and I went with this solution.

**Comment by shalabhms** [comment]
@mukundansundar , could you please review this PR..changes around install/uninstall ?

File: pkg/standalone/standalone.go [hunk 3]
```diff
@@ -612,41 +612,45 @@ func untar(filepath, targetDir, binaryFilePrefix string) (string, error) {
...
+		defer f.Close()
+
+		// #nosec G110
+		if _, err = io.Copy(f, tr); err != nil {
+			return "", err
+		}
 
-			return extractedFilePath, nil
+		// If the found file is the binary that we want to find, save it and return later
+		if strings.HasSuffix(header.Name, binaryFilePrefix) || strings.HasSuffix(header.Name, binaryFilePrefix+".exe") {
```
Comment (mukundansundar) [inline pkg/standalone/standalone.go:649]: runtime.GOOS is windows is handled by unzip method

File: pkg/standalone/standalone.go [hunk 4]
```diff
@@ -612,41 +612,45 @@ func untar(filepath, targetDir, binaryFilePrefix string) (string, error) {
 
 	tr := tar.NewReader(gzr)
 
+	foundBinary := ""
```
Comment (mukundansundar) [inline pkg/standalone/standalone.go:615]: no need for foundBinary ... can be refactored

File: pkg/standalone/uninstall.go [hunk 5]
```diff
@@ -97,7 +96,6 @@ func Uninstall(uninstallAll bool, dockerNetwork string) error {
 		}
 	}
 
-	err = errors.New("uninstall failed")
```
Comment (mukundansundar) [inline pkg/standalone/uninstall.go:100]: Why is this removed?

File: pkg/standalone/standalone.go [hunk 6]
```diff
@@ -429,29 +429,29 @@ func runPlacementService(wg *sync.WaitGroup, errorChan chan<- error, dir, versio
...
 		version = version[1:]
 	}
-	// https://github.com/dapr/dapr/releases/download/v0.8.0/daprd_darwin_amd64.tar.gz
-	// https://github.com/dapr/dapr/releases/download/v0.8.0/placement_darwin_amd64.tar.gz
+
 	fileURL := fmt.Sprintf(
 		"https://github.com/%s/%s/releases/download/v%s/%s_%s_%s.%s",
 		cli_ver.DaprGitHubOrg,
-		cli_ver.DaprGitHubRepo,
+		githubRepo,
```
Comment (mukundansundar) [inline pkg/standalone/standalone.go:454]: why is this a local variable while the Org one is not?

File: pkg/standalone/standalone.go (hunk 3 above)
Comment (willdavsmith) [reply pkg/standalone/standalone.go:649]: The current artifacts published for Dashboard are tarfiles for both windows and linux, So I had to make adjustments to this

File: pkg/standalone/standalone.go (hunk 4 above)
Comment (willdavsmith) [reply pkg/standalone/standalone.go:615]: I added foundBinary to ensure that all files in the tarball are extracted, even if the reader finds the binary first. If I returned as soon as I found the binary, then the function may not extract everything from the tarball.

File: pkg/standalone/standalone.go (hunk 6 above)
Comment (willdavsmith) [reply pkg/standalone/standalone.go:454]: I left the Org as such because all binaries that we install should be from dapr org. I can change this though, just let me know.

**Comment by shalabhms** [comment]
@mukundansundar , could you please look into the comments reply and resolve if looks good?

File: pkg/standalone/standalone.go (hunk 3 above)
Comment (willdavsmith) [reply pkg/standalone/standalone.go]: The artifact for dashboard on windows is now a zip archive, so I updated this.

File: cmd/dashboard.go [hunk 7]
```diff
@@ -49,90 +49,92 @@ var DashboardCmd = &cobra.Command{
...
-		// search for dashboard service namespace in order:
-		// user-supplied namespace, dapr-system, default
-		namespaces := []string{dashboardNamespace}
-		if dashboardNamespace != daprSystemNamespace {
-			namespaces = append(namespaces, daprSystemNamespace)
-		}
-		if dashboardNamespace != defaultNamespace {
-			namespaces = append(namespaces, defaultNamespace)
-		}
+			// search for dashboard service namespace in order:
```
[the remark this answers is not in the document]
Comment (mukundansundar) [reply cmd/dashboard.go:59]: This still looks like tabs. Can you recheck this part?

[the remark this answers is not in the document]
Comment (shalabhms) [reply cmd/dashboard.go:59]: Figured out that Github provides the setting to hide view space change..Once we have that setting , it does not show .
